python_modules/dagster/dagster/components/core/tree.py

```diff
@@ -106,11 +107,11 @@ def load_context(self):
         )
 
     @cached_method
-    def find_root_decl(self) -> DefsFolderDecl:
-        return DefsFolderDecl.get(self.load_context)
+    def find_root_decl(self) -> ComponentDecl:
+        return check.not_none(get_component_decl(self.load_context))
 
     @cached_method
-    def load_root_component(self) -> DefsFolderComponent:
+    def load_root_component(self) -> Component:
```

Comment:
hmm, wonder if we should quarantine off this "legacy" behavior in a different way and preserve these "root" expectations for the normal code paths

like, would a `legacy_build_defs` fn called from `load_defs` allow you to keep this stuff in one cluster of methods ?

Reply:
bummer, inescapable complexity then. Plan is to flesh things out as we go and replace these exceptions with handling more conditions?